src/main.rs

```diff
@@ -53,8 +53,17 @@ fn main() {
                 .short("e")
                 .takes_value(true)
                 .possible_values(&["hyper", "reqwest"])
+                .default_value("hyper")
                 .help("The engine to use"),
         )
+        .arg(
+            Arg::with_name("header")
+                .long("header")
+                .multiple(true)
+                .takes_value(true)
+                .number_of_values(1)
```

Comment:
Doesn't this limit it to only a single header? Can we remove this?

Reply:
No. The way I intended this to work is like so:
`--header <key1>:<val1> --header <key2>:<val2>`

The "1" signifies that only one value will be supplied after the switch. This along with 'multiple(true)' will ensure that we can take in multiple values.

Otherwise, clap won't know where the positional args start and where the optional arg ends. Let me know if you have another idea.

Reply:
Ah, I must have misunderstood how the works. Thanks.